include/Core/JsonUtility.hpp

```diff
@@ -0,0 +1,17 @@
+#ifndef FUG_CORE_JSON_UTILITY_HPP
+#define FUG_CORE_JSON_UTILITY_HPP
+
+#include <unordered_set>
+#include <json/json.hpp>
+
+namespace fug {
+
+// Checks that given json doesn't have invalid keys and contains all required keys
+// Potential errors are printed to FUG_LOG(LogLevel::Error) and result in a throw
+void checkJsonValidity(const std::string& type, const nlohmann::json& json,
```

Comment:
Based on the name, I would have assumed this to return something. This seems to something more of an assert.